nilearn/image/image.py

```diff
@@ -922,6 +922,32 @@ def threshold_img(
     Thresholding can be done based on direct image intensities or selection
     threshold with given percentile.
 
+    If threshold is float, we threshold the image based on image intensities.
+    When `two_sided` is True, the given value should be within the range of
+    minimum and maximum intensity of the input image. All instensities in the
+    interval [-threshold, threshold] will be set to zero.
+
+    When `two_sided` is False:
+
+    If the threshold is negative, then it should be greater than the minimum
+    intensity of the input data. All intensities greater than or equal to the
+    specified threshold will be set to zero.
+
+    If the threshold is positive, then it should be less than the maximum
+    intensity of the input data. All intensities less than or equal to the
+    specified threshold will be set to zero.
+
+    All other instensities keep their original values.
+
+    If threshold is str, the number part should be in interval [0, 100].
+    We threshold the image based on the score obtained using this percentile on
+    the image data. The percentile rank is computed using
+    :func:`scipy.stats.scoreatpercentile`.
+    When `two_sided` is True, score is calculated on the absolute values of
+    data.
+    When `two_sided` is False, score is calculated on only the non-negative
+    values of data.
+
```

Comment:
```suggestion
    - If ``threshold`` is a :obj:`float`,
      we threshold the image based on image intensities.
    
      - When ``two_sided`` is True:
      
        the given value should be
        within the range of minimum and maximum intensity
        of the input image.
        All instensities in the interval ``[-threshold, threshold]``
        will be set to zero.

      - When ``two_sided`` is False:

        - If the threshold is negative,
          then it should be greater than the minimum intensity
          of the input data.
          All intensities greater than or equal
          to the specified threshold will be set to zero.

        - If the threshold is positive,
          then it should be less than the maximum intensity
          of the input data.
          All intensities less than or equal
          to the specified threshold will be set to zero.

       All other instensities keep their original values.

    - If threshold is :obj:`str`,
      the number part should be in interval ``[0, 100]``.
      We threshold the image based on the score
      obtained using this percentile on the image data.
      The percentile rank is computed using
      :func:`scipy.stats.scoreatpercentile`.
      
      - When ``two_sided`` is True, 
        teh score is calculated on the absolute values of data.
      
      - When ``two_sided`` is False, the score is calculated
        only on the non-negative values of data.

```

Reply:
mostly to try to help with the different 'if' blocks

Reply:
I firstly did it with bullets, but docs did not build correctly. May be I did not do it correctly. I am committing this separately, so that if necessary, we can revert it.

Reply:
will check to build the doc locally and push any formatting issue

Reply:
There was a minor issue.

Final rendering should be this:

![image](https://github.com/user-attachments/assets/0fe5e136-8703-4abc-8fbd-89f53a094749)